packages/textlint-rule-ja-space-between-half-and-full-width/src/index.js

```diff
@@ -7,36 +7,46 @@ const assert = require("assert");
 import {RuleHelper} from "textlint-rule-helper";
 import {matchCaptureGroupAll} from "match-index";
 const PunctuationRegExp = /[。、]/;
-const ZenRegExpStr = '[、。]|[\u3400-\u4DBF\u4E00-\u9FFF\uF900-\uFAFF]|[\uD840-\uD87F][\uDC00-\uDFFF]|[ぁ-んァ-ヶ]'
+const ZenRegExpStr = '[、。]|[\u3400-\u4DBF\u4E00-\u9FFF\uF900-\uFAFF]|[\uD840-\uD87F][\uDC00-\uDFFF]|[ぁ-んァ-ヶ]';
+const defaultSpaceOptions = {
+    alphabets: false,
+    numbers: false,
+    punctuation: false
+};
 const defaultOptions = {
-    // スペースを入れるかどうか
-    // "never" or "always"
-    space: "never",
-    // 半角数字を無視するかどうか
-    // only works w/ space: "always"
-    ignoreNumbers: false,
-    // [。、,.]を例外とするかどうか
-    exceptPunctuation: true,
     // プレーンテキスト以外を対象とするか See https://github.com/textlint/textlint-rule-helper#rulehelperisplainstrnodenode-boolean
     lintStyledNode: false,
 };
 function reporter(context, options = {}) {
+   /**
+    * 入力された `space` オプションを内部処理用に成形する
+    * @param {string|Array|undefined} opt `space` オプションのインプット
+    * @param {boolean|undefined} exceptPunctuation `exceptPunctuation` オプションのインプット
+    * @returns {Object}
+    */
+    const parseSpaceOption = (opt, exceptPunctuation) => {
+      let option = {...defaultSpaceOptions};
+
+      if (typeof opt === 'string') {
+        assert(opt === "always" || opt === "never", `"space" options should be "always", "never" or an array.`);
+
+        if (opt === 'always') option = { ...option, alphabets: true, numbers: true };
+        if (exceptPunctuation === false) option = { ...option, punctuation: true };
+      } else if (Array.isArray(opt)) {
+        assert(opt.every((v) => Object.keys(option).includes(v)), `Only "alphabets", "numbers", "punctuation" can be included in the array.`);
+
+        opt.map(key => option[key] = true);
```

Comment:
```suggestion
        const userOptions = Object.fromEntries(opt.map(key => [key, true]));
        return {...defaultoptions, ...userOptions};
```

とすれば、 defaultとuser optionを合成って感じで書けますかね

```js
Object.fromEntries(["a","b","c"].map(key => [key, true]));
// Object { a: true, b: true, c: true }
```

Reply:
こちらで修正しました。
https://github.com/textlint-ja/textlint-rule-preset-ja-spacing/pull/45/commits/4c905e9c8882fe4d0681e1b44f70685e2ea857ba